Build failure on GCC 15: conflicting types for zzpre_ast

I'm getting this build error in both 0.89 and 0.90 between Fedora 41 and Fedora 42:
```
In file included from btparse/src/bibtex.c:36:
btparse/src/../pccts/ast.c:139:1: error: conflicting types for ‘zzpre_ast’; have ‘void(AST *, void (*)(AST *), void (*)(AST *), void (*)(AST *))’ {aka ‘void(struct _ast *, void (*)(struct _ast *), void (*)(struct _ast *), void (*)(struct _ast *))’}
  139 | zzpre_ast(
      | ^~~~~~~~~
In file included from btparse/src/stdpccts.h:28,
                 from btparse/src/parse_auxiliary.h:24,
                 from btparse/src/bibtex.c:21:
btparse/src/../pccts/ast.h:93:6: note: previous declaration of ‘zzpre_ast’ with type ‘void(AST *, void (*)(void), void (*)(void), void (*)(void))’ {aka ‘void(struct _ast *, void (*)(void), void (*)(void), void (*)(void))’}
   93 | void zzpre_ast(AST *, void (*)(), void (*)(), void (*)());
      |      ^~~~~~~~~
```
Here's the full log:
https://kojipkgs.fedoraproject.org//work/tasks/2463/128542463/build.log

On Fedora 41:
```
gcc --version
gcc (GCC) 14.2.1 20250110 (Red Hat 14.2.1-7)
```

On Fedora 42:
```
gcc --version
gcc (GCC) 15.0.1 20250125 (Red Hat 15.0.1-0)
```

Downstream reference: https://bugzilla.redhat.com/show_bug.cgi?id=2341038

I'll post more details if I get any further debugging this, but I thought it might be something you'd recognize faster than me!

--- a/btparse/pccts/ast.h
+++ b/btparse/pccts/ast.h
@@ -90,7 +90,7 @@ extern AST *zzastStack[];
 void zzlink(AST **, AST **, AST **);
 void zzsubchild(AST **, AST **, AST **);
 void zzsubroot(AST **, AST **, AST **);
-void zzpre_ast(AST *, void (*)(), void (*)(), void (*)());
+void zzpre_ast(AST *, void (*)(AST *), void (*)(AST *), void (*)(AST *));
 void zzfree_ast(AST *);
 AST *zztmake(AST *, ...);
 AST *zzdup_ast(AST *);

